compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -167,6 +168,11 @@ struct LoweringContext<'a, 'hir: 'a> {
 
     allow_try_trait: Option<Lrc<[Symbol]>>,
     allow_gen_future: Option<Lrc<[Symbol]>>,
+
+    /// This is `Some` while lowering the Pat of a `let-else`. Any bindings in the pattern are given
+    /// a new `HirId`. The lowered `HirId` of the binding is added here and then used to create
+    /// bindings in the outer `let` statement of the desugared `let-else`.
+    let_else_bindings: Option<FxIndexMap<hir::HirId, (hir::BindingAnnotation, hir::HirId, Ident)>>,
```

Comment:
I am not really fond of having an extra global variable during lowering. Is there an easy way to do it without?

Reply:
I don't think so. I would have to pass this structure to every `lower_pat_*` method.